GL-Work

romilqlik opened the pull request:
# Description of change 


This is Combined PR of:

- [ ] [SAC-27869](https://github.com/singer-io/tap-listrak/pull/19) - fix libraries and dependabot issues
- [ ] [SAC-27861](https://github.com/singer-io/tap-listrak/pull/19) -  5XX and Unhandled Listrak exception handling and Schema Updates

# Manual QA steps

- [ ]  Discovery: Working
- [ ]  Sync: NA (Test Account not available)
- [ ]  Unittests: passing
- [ ]  Integration test: not implemented

# Risks

- [ ] NA


#Rollback steps

- [ ] Revert this branch

RushiT0122 reviewed (changes requested):
On my local testing I observed couple of issue,

1. `[Logon] [InvalidLogonAttempt] Invalid Credentials Provided.` is getting retried 5 times, shouldn't be retried.
2. [sync implementation](https://github.com/singer-io/tap-listrak/blob/gl-master/tap_listrak/__init__.py#L64) needs to be revisited.

RushiT0122 reviewed (changes requested):
Couple observations needs the attenssion,

1. IMO, InvalidLogonAttempt shouldn't be retried.
```
ERROR Giving up request(...) after 5 tries (zeep.exceptions.Fault: GetContactListCollection(): [Logon] [InvalidLogonAttempt] Invalid Credentials Provided)
```
2. [sync implementation](https://github.com/singer-io/tap-listrak/blob/gl-master/tap_listrak/__init__.py#L64) needs to be revisited.

romilqlik commented on setup.py:
```diff
@@ -10,11 +10,13 @@
-        "zeep",
-        'backoff==1.8.0',
-        'pendulum==1.2.0',
+        "singer-python==6.1.1",
```
This is done.

romilqlik commented:
This is done.
in http.py : Updated the requests() function to skip retries for Fault exceptions that contain "InvalidLogonAttempt".

in __init__.py: Refactored sync() function to dynamically call the sync function for each selected stream insted of hardcoding sync_lists(),

RushiT0122 commented on tap_listrak/http.py:
```diff
@@ -17,23 +16,36 @@ def get_client(config):
+    max_tries=5,
+    jitter=None,
+    on_backoff=log_retry_attempt,
+    giveup=is_non_retriable_exception # This will prevent retry on bad login
```
Since the giveup handler is named `is_non_retriable_exception()`, the comment doesn't add any value and seems redundant. IMO, no comment is needed here.

RushiT0122 replied:
Also please avoid force pushes.
